client.py

```diff
@@ -1,53 +1,67 @@
-#
-#   Request-reply client in Python
-#   Connects REQ socket to tcp://localhost:5559
-#   Sends "Hello" to server, expects "World" back
-#
 import tqdm as tqdm
 import zmq
 import numpy as np
 from utils import send_array
 from numpy import random
 
 
+class Client:
+    def __init__(self, addr="tcp://localhost:5559", port=8192):
+        self.context = zmq.Context()
+        self.socket = self.context.socket(zmq.DEALER)
+
+        self.socket.set(zmq.SNDHWM, port)
+        self.socket.set(zmq.RCVHWM, port)
+        self.socket.connect(addr)
+
+        self.poller = zmq.Poller()
+        self.poller.register(self.socket, zmq.POLLIN | zmq.POLLOUT)
```

Comment:
You have to register the socket on each call to `compute`, because `compute` can be called multiple times and if you didn't reset it, it wouldn't have `POLLOUT`.